wkhtmltopdf-image/src/main/docker/app.py

```diff
@@ -0,0 +1,138 @@
+#! /usr/bin/env python
+import StringIO
+import json
+import logging
+import uuid
+from os import EX_OK
+from subprocess import Popen, PIPE
+
+from werkzeug.exceptions import HTTPException, NotFound
+from werkzeug.routing import Map, Rule
+from werkzeug.wrappers import Request, Response
+
+"""
+This python server is responsible for generating PDFs from HTML
+using an underlying wkhtmltopdf command-line tool.
+
+To use this application, the user must send a POST request with
+base64 or form encoded encoded HTML content and the wkhtmltopdf Options in
+request data, with keys 'contents' and 'options'.
+The application will return a response with the PDF file.
+"""
+class WkHtmlToPdfServer():
+    def __init__(self):
+        self.url_map = Map([
+            Rule('/', endpoint='generate_pdf'),
+            Rule('/health', endpoint='health_check')
+        ])
+        logging.basicConfig()
+        self.logger = logging.getLogger('WkHtmlToPdfServer')
+        #TODO allow environment to configure this
+        self.logger.setLevel(logging.INFO)
+
+    """
+    Generate a PDF from submitted HTML.
+    """
+    def on_generate_pdf(self, request):
+        if request.method != 'POST':
+            return
+
+        request_is_json = request.content_type.endswith('json')
+        sourceStream = None
+        options = None
+
+        traceId = request.headers.get('X-Trace-Id', default=uuid.uuid4())
+        self.logger.info("Received PDF request: %s", traceId)
```

Comment:
Hopefully whoever's requesting the pdf will send us an X-Trace-Id header so we can track cross-service requests in the logs.